backend/index.js

```diff
@@ -0,0 +1,99 @@
+const express = require("express");
+const app = express();
+const orm = require("sequelize");
+const database = new orm.Sequelize({
+  dialect: "sqlite",
+  storage: "./data/catsDatabase.sqlite",
+});
+
+const Cat = database.define("Cat", {
+  name: {
+    type: orm.DataTypes.STRING,
+    allowNull: false,
+  },
+  imageUrl: {
+    type: orm.DataTypes.STRING,
+    allowNull: false,
+  },
+  age: {
+    type: orm.DataTypes.INTEGER,
+    allowNull: false,
+  },
+  description: {
+    type: orm.DataTypes.STRING,
+  },
+  available: {
+    type: orm.DataTypes.BOOLEAN,
+    defaultValue: true,
+  },
+});
+
+async function createCatsHandler(req, res) {
+  if (
+    typeof req.body.name === "string" &&
+    req.body.name.length > 0 &&
+    typeof req.body.imageUrl === "string" &&
+    req.body.imageUrl.length > 0 &&
+    typeof req.body.age === "number" &&
+    req.body.age > 0 &&
+    typeof req.body.description !== "undefined"
+  ) {
+    const newCat = await Cat.create(req.body);

```

Comment:
Never pass directly the whole body object "req.body"

Risk of passing unwanted props

Prefer deconstruction pattern for exemple :

```javascript
const {name, imageUrl, age, description} = req.body
const newCat = await Cat.create({ name, imageUrl, age, description});
```

or

```javascript
const newCat = await Cat.create({name: req.body.name, description: req.body.description // etc...})
```